core/indigo-core/common/math/algebra.h

```diff
@@ -822,5 +824,234 @@ namespace indigo
         float _d;
     };
 
+    inline bool isPointInPolygon(const Vec2f& p, const std::vector<Vec2f>& poly)
+    {
+        bool in = false;
+        for (size_t i = 0, n = poly.size(); i < n; ++i)
+        {
+            size_t j = (i + 1) % n;
+            bool intersect =
+                ((poly[i].y > p.y) != (poly[j].y > p.y)) && (p.x < (poly[j].x - poly[i].x) * (p.y - poly[i].y) / (poly[j].y - poly[i].y) + poly[i].x);
+            if (intersect)
+                in = !in;
+        }
+        return in;
+    }
+
+    inline std::vector<Vec2f> getPointsInsidePolygon(const std::vector<Vec2f>& polygon1, const std::vector<Vec2f>& polygon2)
+    {
+        std::vector<Vec2f> result;
+        result.reserve(polygon1.size());
+        std::copy_if(polygon1.begin(), polygon1.end(), std::back_inserter(result), [&](auto& p) { return isPointInPolygon(p, polygon2); });
+        return result;
+    }
+
+    inline bool isPointInConvexPolygon(const Vec2f& p, const std::vector<Vec2f>& poly)
```

Comment:
Looks like this function computation is more complex tnan isPointInPolygon.
For convex polygon in most cases no edge crossing computation need - edge will be at left or at right from point.

Reply:
This function has an "early exit" in comparison to ray-casting approach. So in some cases it works faster.